packages/esm-ward-app/src/ward-view/ward-view.component.tsx

```diff
@@ -45,20 +47,55 @@ const WardView = () => {
 };
 
 const WardViewByLocation = ({ location }: { location: Location }) => {
-  const { admissionLocation, isLoading, error } = useAdmissionLocation(location.uuid);
+  const {
+    admissionLocation,
+    isLoading: isLoadingLocation,
+    error: errorLoadingLocation,
+  } = useAdmissionLocation(location.uuid);
+  const {
+    admittedPatients,
+    isLoading: isLoadingPatients,
+    error: errorLoadingPatients,
+  } = useAdmittedPatients(location.uuid);
   const { t } = useTranslation();
+  const admittedPatientsByUuid = useMemo(() => {
+    const map = new Map<string, AdmittedPatient>();
+    for (const admittedPatient of admittedPatients ?? []) {
+      map.set(admittedPatient.patient.uuid, admittedPatient);
+    }
+    return map;
+  }, [admittedPatients]);
 
-  if (admissionLocation) {
+  if (admissionLocation != null && admittedPatients != null) {
     const bedLayouts = filterBeds(admissionLocation);
 
     return (
       <>
         {bedLayouts.map((bedLayout, i) => {
           const { patients } = bedLayout;
           const bed = bedLayoutToBed(bedLayout);
+          const patientInfos: WardPatient[] = patients.map((patient) => {
```

Comment:
This is not a blocker but it would be much more idiomatic to put all this (from `bedLayouts.map(` to `return <WardBed`) into a `useMemo` block and just render the result of that in this return block. This keeps computation logic and rendering logic separate, which makes for more idiomatic and readable React code.

Reply:
I can't easily do `useMemo` as this is in an if statement, but I can store this in a variable instead of inlining it in the JSX.